### PR TITLE
Sbachmei/mic 5125/import subpackages to top level namespace

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  other
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5125

### Changes and notes
 This simply imports most of the classes to the top-level namespace
so that the user doesn't need to know the full filepaths.
Note:

- I only imported classes (not functions or constants) except for a few instances
where something else was already being imported at the sub-package level
- I did not include anything from the mslt subpackage
- I didi not include error classes

### Testing

tests pass; I confirmed a few imports

### CODE REVIEW

File: src/vivarium_public_health/__init__.py [hunk 1]
```diff
@@ -28,6 +28,18 @@
...
+from vivarium_public_health.population import (
+    AgeOutSimulants,
+    BasePopulation,
+    FertilityAgeSpecificRates,
+    FertilityCrudeBirthRate,
+    FertilityDeterministic,
+    Mortality,
+    generate_population,
+)
+from vivarium_public_health.results import (
```
Comment (hussain-jafari) [inline src/vivarium_public_health/__init__.py:40]: Should this be empty?

**Review by hussain-jafari — APPROVED** [review]

File: src/vivarium_public_health/__init__.py (hunk 1 above)
Comment (stevebachmeier) [reply src/vivarium_public_health/__init__.py:40]: Not sure what you mean. It's importing a bunch of things? lines 41-48. @hussain-jafari

Comment (hussain-jafari) [reply src/vivarium_public_health/__init__.py:40]: Oh odd it looks empty when I look commit by commit but looking at the whole PR I see that it has a bunch of things being imported.

**Review by patricktnast — APPROVED** [review]

File: src/vivarium_public_health/__init__.py (hunk 1 above)
Comment (stevebachmeier) [reply src/vivarium_public_health/__init__.py:40]: Ah, I might have accidentally pushed an empty list in the wrong commit. Good eye!

**Review by rmudambi — COMMENTED** [review]
I'm not sure we want everything available at the top level. I think we should have a conversation about this.

**Pull request merged**